liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -0,0 +1,889 @@
+import static org.twdata.maven.mojoexecutor.MojoExecutor.plugin;
+import static org.twdata.maven.mojoexecutor.MojoExecutor.version;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Paths;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+import org.apache.maven.model.PluginExecution;
+import org.apache.maven.model.Resource;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.custommonkey.xmlunit.DetailedDiff;
+import org.custommonkey.xmlunit.XMLUnit;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.resolution.DependencyResolutionException;
+import org.eclipse.aether.resolution.DependencyResult;
+import org.twdata.maven.mojoexecutor.MojoExecutor.Element;
+
+import net.wasdev.wlp.ant.ServerTask;
+import net.wasdev.wlp.common.plugins.util.DevUtil;
+import net.wasdev.wlp.common.plugins.util.ServerFeatureUtil;
+
+/**
+ * Start a liberty server in dev mode import to set ResolutionScope for TEST as
+ * it helps build full transitive dependency classpath
+ */
+@Mojo(name = "dev", requiresDependencyCollection = ResolutionScope.TEST, requiresDependencyResolution = ResolutionScope.TEST)
+public class DevMojo extends StartDebugMojoSupport {
+
+    private static final String UPDATED_APP_MESSAGE_REGEXP = "CWWKZ0003I.*";
+    
+    @Parameter(property = "hotTests", defaultValue = "false")
+    private boolean hotTests;
+
+    @Parameter(property = "skipTests", defaultValue = "false")
+    private boolean skipTests;
+
+    @Parameter(property = "skipUTs", defaultValue = "false")
+    private boolean skipUTs;
+
+    @Parameter(property = "skipITs", defaultValue = "false")
+    private boolean skipITs;
+
+    @Parameter(property = "liberty.debug.port", defaultValue = "7777")
```

Comment:
Noticed we don't have a similar parameter for the debug goal. Is this something we should consider adding later?

Reply:
For the debug goal, it's expected that the user manually edits server.env and sets the port in there.  It should use exactly what is provided in server.env.  That matches Liberty's `server debug` behaviour.  

For dev mode, debug is provided as an additional convenience, so the `dev` goal accepts the additional `liberty.debug.port` property and helps set it in server.env for the user.